### PR TITLE
zerocopy: Some Ref methods are unsound with some type params

### PR DESCRIPTION
**Pull request by joshlf**
For more information:
- https://github.com/google/zerocopy/issues/716
- https://github.com/google/zerocopy/security/advisories/GHSA-3mv5-343c-w2qg

### CODE REVIEW

**Comment by joshlf** [comment]
Re: CI failure: Is there a way to omit the version specifier for each function? All listed functions are affected in all crate versions affected by this advisory.

**Comment by tarcieri** [comment]
Unfortunately not, although once you write it once you can copy/paste

**Comment by joshlf** [comment]
@tarcieri @Shnatsel Sorry to nag, but we need this advisory to go out before we can yank the affected versions.

File: crates/zerocopy/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+unaffected = ["< 0.2.2"]
+
+[affected.functions]
+"zerocopy::Ref::into_ref" = [">= 0.2.2, < 0.2.9", ">= 0.3.0, < 0.3.2", ">= 0.4.0, < 0.4.1", ">= 0.5.0, < 0.5.2", ">= 0.6.0, < 0.6.6", ">= 0.7.0, < 0.7.31"]
+"zerocopy::Ref::into_mut" = [">= 0.2.2, < 0.2.9", ">= 0.3.0, < 0.3.2", ">= 0.4.0, < 0.4.1", ">= 0.5.0, < 0.5.2", ">= 0.6.0, < 0.6.6", ">= 0.7.0, < 0.7.31"]
+"zerocopy::Ref::into_slice" = [">= 0.2.2, < 0.2.9", ">= 0.3.0, < 0.3.2", ">= 0.4.0, < 0.4.1", ">= 0.5.0, < 0.5.2", ">= 0.6.0, < 0.6.6", ">= 0.7.0, < 0.7.31"]
+"zerocopy::Ref::into_mut_slice" = [">= 0.2.2, < 0.2.9", ">= 0.3.0, < 0.3.2", ">= 0.4.0, < 0.4.1", ">= 0.5.0, < 0.5.2", ">= 0.6.0, < 0.6.6", ">= 0.7.0, < 0.7.31"]
+```
+
+# RustSec Advisory Template - Advisory Title Goes Here
```
Comment (tarcieri) [inline crates/zerocopy/RUSTSEC-0000-0000.md:20]: Needs an advisory title

Comment (joshlf) [reply crates/zerocopy/RUSTSEC-0000-0000.md]: Done.

**Review by tarcieri — APPROVED** [review]

**Comment by tarcieri** [comment]
Assigned `RUSTSEC-2023-0074` in #1839

**Comment by joshlf** [comment]
Thanks for the review, @tarcieri! And apologies again about nagging.

**Pull request merged**